pymysqlreplication/binlogstream.py

```diff
@@ -748,14 +749,12 @@ def _allowed_event_list(
     def __get_dbms(self):
         if not self.__connected_ctl:
             self.__connect_to_ctl()
-
-        cur = self._ctl_connection.cursor()
-        cur.execute("SELECT VERSION();")
-
-        version_info = cur.fetchone().get("VERSION()", "")
-
-        if "MariaDB" in version_info:
+        if self.dbms:
+            return self.dbms
+        if "MariaDB" in self._ctl_connection.get_server_info():
```

Comment:
really Cool!
We can reduce the master load a bit more.

Reply:
Thanks !!